Index: busybox-1.29.3/networking/httpd.c
===================================================================
--- busybox-1.29.3.orig/networking/httpd.c
+++ busybox-1.29.3/networking/httpd.c
@@ -1019,13 +1019,14 @@ static void log_and_exit(void)
 {
 	/* Paranoia. IE said to be buggy. It may send some extra data
 	 * or be confused by us just exiting without SHUT_WR. Oh well. */
-	shutdown(1, SHUT_WR);
-	/* Why??
-	(this also messes up stdin when user runs httpd -i from terminal)
-	ndelay_on(0);
-	while (read(STDIN_FILENO, iobuf, IOBUF_SIZE) > 0)
+	int t;
+	socklen_t tl = sizeof(t);
+	if (getsockopt(STDIN_FILENO, SOL_SOCKET, SO_TYPE, &t, &tl) == 0)
+		shutdown(STDIN_FILENO, SHUT_WR);
+	/* Eat any remaining data for a while to allow response to be seen */
+	alarm(30);
+	while (recv(STDIN_FILENO, iobuf, IOBUF_SIZE, MSG_DONTWAIT) > 0)
 		continue;
-	*/
 
 	if (verbose > 2)
 		bb_error_msg("closed");
