Index: busybox-1.29.3/networking/ntpd.c
===================================================================
--- busybox-1.29.3.orig/networking/ntpd.c
+++ busybox-1.29.3/networking/ntpd.c
@@ -76,6 +76,7 @@
 //usage:     "\n	-q	Quit after clock is set"
 //usage:     "\n	-N	Run at high priority"
 //usage:     "\n	-w	Do not set time (only query peers), implies -n"
+//usage:     "\n	-s	Log to syslog, even if -n is given"
 //usage:     "\n	-S PROG	Run PROG after stepping time, stratum change, and every 11 mins"
 //usage:     "\n	-p PEER	Obtain time from PEER (may be repeated)"
 //usage:	IF_FEATURE_NTPD_CONF(
@@ -93,6 +94,7 @@
 
 #include "libbb.h"
 #include <math.h>
+#include <syslog.h>
 #include <netinet/ip.h> /* For IPTOS_DSCP_AF21 definition */
 #include <sys/timex.h>
 #ifndef IPTOS_DSCP_AF21
@@ -343,6 +345,7 @@ enum {
 	OPT_S = (1 << 6),
 	OPT_l = (1 << 7) * ENABLE_FEATURE_NTPD_SERVER,
 	OPT_I = (1 << 8) * ENABLE_FEATURE_NTPD_SERVER,
+	OPT_s = (1 << 9),
 	/* We hijack some bits for other purposes */
 	OPT_qq = (1 << 31),
 };
@@ -2297,6 +2300,7 @@ static NOINLINE void ntp_init(char **arg
 			"nqNx" /* compat */
 			"wp:*S:"IF_FEATURE_NTPD_SERVER("l") /* NOT compat */
 			IF_FEATURE_NTPD_SERVER("I:") /* compat */
+			"s"    /* NOT compat */
 			"d" /* compat */
 			"46aAbgL" /* compat, ignored */
 				"\0"
@@ -2332,6 +2336,12 @@ static NOINLINE void ntp_init(char **arg
 		logmode = LOGMODE_NONE;
 	}
 
+	/* Regardless of OPT_n we enable syslog here */
+	if (opts & OPT_s) {
+		openlog(applet_name, LOG_PID, LOG_DAEMON);
+		logmode = LOGMODE_SYSLOG;
+	}
+
 	if (peers) {
 		while (peers)
 			add_peers(llist_pop(&peers));
