Index: busybox-1.23.2/networking/ntpd.c
===================================================================
--- busybox-1.23.2.orig/networking/ntpd.c
+++ busybox-1.23.2/networking/ntpd.c
@@ -29,11 +29,12 @@
  */
 
 //usage:#define ntpd_trivial_usage
-//usage:	"[-dnqNw"IF_FEATURE_NTPD_SERVER("l -I IFACE")"] [-S PROG] [-p PEER]..."
+//usage:	"[-dnsqNw"IF_FEATURE_NTPD_SERVER("l -I IFACE")"] [-S PROG] [-p PEER]..."
 //usage:#define ntpd_full_usage "\n\n"
 //usage:       "NTP client/server\n"
 //usage:     "\n	-d	Verbose"
 //usage:     "\n	-n	Do not daemonize"
+//usage:     "\n	-s	Log to syslog, even if -n is given"
 //usage:     "\n	-q	Quit after clock is set"
 //usage:     "\n	-N	Run at high priority"
 //usage:     "\n	-w	Do not set time (only query peers), implies -n"
@@ -53,6 +54,7 @@
 
 #include "libbb.h"
 #include <math.h>
+#include <syslog.h>
 #include <netinet/ip.h> /* For IPTOS_LOWDELAY definition */
 #include <sys/resource.h> /* setpriority */
 #include <sys/timex.h>
@@ -293,6 +295,7 @@ enum {
 	OPT_S = (1 << 6),
 	OPT_l = (1 << 7) * ENABLE_FEATURE_NTPD_SERVER,
 	OPT_I = (1 << 8) * ENABLE_FEATURE_NTPD_SERVER,
+	OPT_s = (1 << 9),
 	/* We hijack some bits for other purposes */
 	OPT_qq = (1 << 31),
 };
@@ -2128,6 +2131,7 @@ static NOINLINE void ntp_init(char **arg
 			"nqNx" /* compat */
 			"wp:S:"IF_FEATURE_NTPD_SERVER("l") /* NOT compat */
 			IF_FEATURE_NTPD_SERVER("I:") /* compat */
+			"s"    /* NOT compat */
 			"d" /* compat */
 			"46aAbgL", /* compat, ignored */
 			&peers,&G.script_name,
@@ -2182,6 +2186,11 @@ static NOINLINE void ntp_init(char **arg
 		bb_daemonize_or_rexec(DAEMON_DEVNULL_STDIO, argv);
 		logmode = LOGMODE_NONE;
 	}
+	/* Regardless of OPT_n we enable syslog here */
+	if (opts & OPT_s) {
+		openlog(applet_name, LOG_PID, LOG_DAEMON);
+		logmode = LOGMODE_SYSLOG;
+	}
 	/* I hesitate to set -20 prio. -15 should be high enough for timekeeping */
 	if (opts & OPT_N)
 		setpriority(PRIO_PROCESS, 0, -15);
