Index: finit-1.8/plugins/initctl.c
===================================================================
--- finit-1.8.orig/plugins/initctl.c	2015-01-18 19:04:25.580318745 +0100
+++ finit-1.8/plugins/initctl.c	2015-01-18 19:04:25.580318745 +0100
@@ -26,6 +26,7 @@
 #include <fcntl.h>		/* O_RDONLY et al */
 #include <signal.h>
 #include <string.h>
+#include <sys/stat.h>
 #include <unistd.h>		/* read() */
 #include <sys/stat.h>
 
