Index: finit-3.0-rc1/finit.c
===================================================================
--- finit-3.0-rc1.orig/finit.c
+++ finit-3.0-rc1/finit.c
@@ -232,7 +232,6 @@ static void emergency_shell(void)
 
 int main(int argc, char* argv[])
 {
-	int err;
 	char *devfsd;
 	uev_ctx_t loop;
 
@@ -362,11 +361,7 @@ int main(int argc, char* argv[])
 	plugin_run_hooks(HOOK_ROOTFS_UP);
 
 	umask(0);
-	print_desc("Mounting filesystems", NULL);
-
-	err = run("/bin/mount -na");
-	print_result(err);
-	if (err)
+	if (run_interactive("/bin/mount -na", "Mounting filesystems"))
 		plugin_run_hooks(HOOK_MOUNT_ERROR);
 
 	run("/sbin/swapon -ea");
