Index: finit-3.0-rc1/finit.c
===================================================================
--- finit-3.0-rc1.orig/finit.c
+++ finit-3.0-rc1/finit.c
@@ -310,6 +310,9 @@ int main(int argc, char* argv[])
 	 */
 	conf_parse_cmdline();
 
+	/* Reduce default kernel log level, only errors or worse, not warnings */
+	system("echo '3 4 1 7' > /proc/sys/kernel/printk");
+
 	/*
 	 * Populate /dev and prepare for runtime events from kernel.
 	 */
