Index: tetris-1.2.1/Makefile
===================================================================
--- tetris-1.2.1.orig/Makefile
+++ tetris-1.2.1/Makefile
@@ -22,7 +22,7 @@ distclean: clean
 install: all
 	@install -D -m 0755 tetris $(DESTDIR)/bin/tetris
 	@touch /tmp/tetris.scores
-	@install -D -m 0664 -g games /tmp/tetris.scores $(DESTDIR)/var/games/tetris.scores
+	@install -D /tmp/tetris.scores $(DESTDIR)/var/games/tetris.scores
 
 dist:
 	@git archive --format=tar --prefix=tetris-$(VERSION)/ $(VERSION) | bzip2 >../tetris-$(VERSION).tar.bz2
