Index: dropbear-2014.65/svr-main.c
===================================================================
--- dropbear-2014.65.orig/svr-main.c	2014-09-17 14:48:38.346608413 +0200
+++ dropbear-2014.65/svr-main.c	2014-09-17 14:50:02.814605425 +0200
@@ -362,8 +362,9 @@
 
 /* catch any segvs */
 static void sigsegv_handler(int UNUSED(unused)) {
-	fprintf(stderr, "Aiee, segfault! You should probably report "
-			"this as a bug to the developer\n");
+        /* Silence and let process supervisor restart dropbear. --Westermo */
+//	fprintf(stderr, "Aiee, segfault! You should probably report "
+//			"this as a bug to the developer\n");
 	_exit(EXIT_FAILURE);
 }
 
