Index: u-boot-2017.07/Makefile
===================================================================
--- u-boot-2017.07.orig/Makefile
+++ u-boot-2017.07/Makefile
@@ -1535,6 +1535,10 @@ distclean: mrproper
 		-type f -print | xargs rm -f
 	@rm -f boards.cfg
 
+PHONY += install
+install:
+	@true
+
 backup:
 	F=`basename $(srctree)` ; cd .. ; \
 	gtar --force-local -zcvf `LC_ALL=C date "+$$F-%Y-%m-%d-%T.tar.gz"` $$F
