From ed88469276c569acd9ef16bcc38cd80afd8dea52 Mon Sep 17 00:00:00 2001
From: Joachim Wiberg <troglobit@gmail.com>
Date: Sun, 10 Dec 2023 17:32:19 +0100
Subject: [PATCH 2/2] Actually silence the log message, missing hunk for
 previous change

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 src/svc.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/svc.c b/src/svc.c
index 6bb90049..773f0ab6 100644
--- a/src/svc.c
+++ b/src/svc.c
@@ -759,11 +759,13 @@ int svc_ifthen(int is_conf, const char *ident, char *stmt, int quiet)
 
 		svc = svc_find_by_str(stmt);
 		if (not && svc) {
-			logit(LOG_NOTICE, "skipping %s, %s already loaded.", ident, svc_ident(svc, NULL, 0));
+			if (!quiet)
+				logit(LOG_NOTICE, "skipping %s, %s already loaded.", ident, svc_ident(svc, NULL, 0));
 			return 0;
 		}
 		if (!not && !svc) {
-			logit(LOG_NOTICE, "skipping %s, %s not available.", ident, stmt);
+			if (!quiet)
+				logit(LOG_NOTICE, "skipping %s, %s not available.", ident, stmt);
 			return 0;
 		}
 	}
-- 
2.34.1

